File: Python/Problems/Problems/problem1523.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set
from collections import Counter

from .problem import Problem


class Problem1523(Problem):
    def test(self) -> bool:
        low = 3
        high = 7

        expected = 3

        result = self.countOdds(low, high)

        print(result)

        return expected == result

    def countOdds(self, low: int, high: int) -> int:
        l = high - low + 1
        if l % 2 == 0:
            return l // 2
        if low % 2 == 1:
            return (l + 1) // 2
        return (l - 1) // 2

File: C++/Problems/Problems/problem_1523.cpp
#include "problem_1523.h"

bool problem_1523::test() {
    int low = 3;
    int high = 7;

    int expected = 3;

    auto result = countOdds(low, high);

    std::cout << result << std::endl;

    return expected == result;
}

int problem_1523::countOdds(int low, int high) {
    int l = high - low + 1;
    if (l % 2 == 0)
        return l / 2;
    if (low % 2 == 1)
        return (l + 1) / 2;
    return (l - 1) / 2;
}
